backend/history_service/controllers/historyController.js

```diff
@@ -0,0 +1,159 @@
+const { db, FieldValue } = require("../config/firebaseConfig");
+const moment = require("moment-timezone");
+
+const getAllAttemptedQuestions = async (req, res) => {
+    const { uid } = req.body;
+
+    try {
+        const usersRef = db.collection("users").doc(uid);
+        const userDoc = await usersRef.get();
+        
+        if (!userDoc.exists) {
+            return res.status(404).json({ message: "User not found" });
+        }
+
+        // Retrieve and filter questionsAttempted array
+        let questionsAttempted = userDoc.data().questionsAttempted || [];
+        const cutoffDate = moment().tz("Asia/Singapore").subtract(180, 'days'); // 180 days ago in Singapore time
```

Comment:
amazing work 🔥 might have to test the api again after the other branches merge in